spec/models/access_request_spec.rb

```diff
@@ -75,4 +75,27 @@
       expect(AccessRequest.by_request_date.last).to eq access_request1
     end
   end
+
+  describe '#add_additonal_attributes' do
+    let(:user) { create(:user, organisations: [organisation]) }
+    let(:organisation) { build(:organisation) }
+    let(:access_request) {
+      build(:access_request,
+            organisation: user.organisations.first.name,
+                requester_email: user.email,
+                requester: nil,
+                request_date_utc: nil,
+                status: nil)
+    }
+
+    before do
+      access_request.add_additonal_attributes(access_request.requester_email)
+    end
+
+    subject { access_request }
+
+    its(:requester)         { should eq user }
+    its(:request_date_utc)  { should be_within(1.second).of Time.now.utc }
```

Comment:
I'd use Timecop around this whole describe block to pin the time to a particular timestamp; then you should be able to do an exact match.

Reply:
Thanks. Have added it 👍